### PR TITLE
feat: add TypeCastExpr class

### PR DESCRIPTION
**Pull request by apkrelling**
## Pull Request description

This PR adds TypeCastExpr class.
This PR is an attempt to solve #110 .

- [X] added new ASTKind's
- [X] created TypeCastExpr classes
- [X] updated `__init__.py` to include the new imports and classes in the`__all__` list
- [X] added the new class visit method in transpilers/python.py
- [X] added test in tests/transpilers/test_python.py
- [X] added test in tests/test_types.py

## How to test these changes
```python
from astx.datatypes import Int32
from astx.types import TypeCastExpr
from astx.variables import Variable
from astx.transpilers import python as astx2py 

# Expression to cast
expr = Variable(name="x")

# Target type for casting
target_type = Int32()

# Create the TypeCastExpr
cast_expr = TypeCastExpr(expr=expr, target_type=target_type)

cast_expr.get_struct()
cast_expr

# Initialize the generator
generator = astx2py.ASTxPythonTranspiler()

# Generate Python code
generated_code = generator.visit(cast_expr)
print(generated_code)
```

## Pull Request checklists

This PR is a:

- [ ] bug-fix
- [X] new feature
- [ ] maintenance

About this PR:

- [X] it includes tests.
- [X] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [X] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more
      complexity.
- [ ] New and old tests passed locally.

### CODE REVIEW

**Comment by apkrelling** [comment]
TO-DO:
- maybe add more tests
- maybe add more transpiler tests
- verify the AST representation in png and ascii (below)

**Comment by apkrelling** [comment]
ASTs2

File: tests/transpilers/test_python.py [hunk 1]
```diff
@@ -371,3 +371,24 @@ def test_literal_complex64() -> None:
...
+    target_type = astx.Int32()
+    # Create the TypeCastExpr
+    cast_expr = astx.TypeCastExpr(expr=expr, target_type=target_type)
+
+    # Initialize the generator
+    generator = astx2py.ASTxPythonTranspiler()
+
+    # Generate Python code
+    generated_code = generator.visit(cast_expr)
+    expected_code = "cast(Int32, x)"
```
Comment (xmnlab) [inline tests/transpilers/test_python.py:390]:
```suggestion
    expected_code = "cast(int, x)"
```

File: src/astx/transpilers/python.py [hunk 2]
```diff
@@ -184,6 +184,11 @@ def visit(self, node: astx.LambdaExpr) -> str:
         params_str = ", ".join(param.name for param in node.params)
         return f"lambda {params_str}: {self.visit(node.body)}"
 
+    @dispatch  # type: ignore[no-redef]
+    def visit(self, node: astx.TypeCastExpr) -> str:
+        """Handle TypeCastExpr nodes."""
+        return f"cast({node.target_type.__class__.__name__}, {node.expr.name})"
```
Comment (xmnlab) [inline src/astx/transpilers/python.py:190]:
```suggestion
        return f"cast({self.visit(node.target_type)}, {node.expr.name})"
```

**Comment by xmnlab** [comment]
I will fix the linter issue in a follow up PR

**Comment by xmnlab** [comment]
thanks for working on that @apkrelling !

**Pull request merged**